### PR TITLE
k8sclusterreceiver: Support OpenShift cluster quota metrics

### PR DESCRIPTION
**Pull request by owais**
**Description:** 
Adds support for OpenShift's cluster resource quota by recording quota spec and usage as metrics.

**Testing:**
- Manually testing with a local OpenShift cluster with CodeReady Containers
- Added tests to the receiver and internal utils.

**Documentation:**
Updated README with receiver config and cluster role examples.

### CODE REVIEW

File: receiver/k8sclusterreceiver/internal/collection/resourcequotas.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func getMetricsForResourceQuota(rq *corev1.ResourceQuota) []*resourceMetrics {
 				&metricspb.Metric{
 					MetricDescriptor: t.metric,
 					Timeseries: []*metricspb.TimeSeries{
-						utils.GetInt64TimeSeriesWithLabels(val, []*metricspb.LabelValue{{Value: string(k)}}),
+						utils.GetInt64TimeSeriesWithLabels(val, []*metricspb.LabelValue{{Value: string(k), HasValue: true}}),
```
Comment (owais) [inline receiver/k8sclusterreceiver/internal/collection/resourcequotas.go:75]: oppurtunistic bug fix. labels were not actually recorded for this timeseries.

File: receiver/k8sclusterreceiver/README.md [hunk 2]
```diff
@@ -265,4 +269,29 @@ spec:
...
+### OpenShift
+
+You can enable OpenShift support to collect OpenShift specific metrics in addition to the default
+k8s ones. To do this, set the OpenShift key in config to `true` to enable OpenShift support.
+
+Example:
+
+```yaml
+  k8s_cluster:
+    openshift: true
```
Comment (jrcamp) [inline receiver/k8sclusterreceiver/README.md:283]: What about maybe making the distribution configurable?

```yaml
k8s_cluster:
  distribution: openshift
```

There might be other features added for other distros.

File: receiver/k8sclusterreceiver/internal/collection/clusteresourcequotas.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	quotav1 "github.com/openshift/api/quota/v1"
+	"go.opentelemetry.io/collector/translator/conventions"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/internal/utils"
+)
+
+var clusterResourceQuotaLimitMetric = &metricspb.MetricDescriptor{
```
Comment (jrcamp) [inline receiver/k8sclusterreceiver/internal/collection/clusteresourcequotas.go:29]: At the time this receiver was written metadata.yaml and pdata weren't around. Would it be possible to add net-new stuff using the new ways so there's less code that has to be translated in the future?

File: workspace.code-workspace [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (jrcamp) [inline workspace.code-workspace:1]: Don't think you meant to commit this.

File: receiver/k8sclusterreceiver/internal/collection/clusteresourcequotas.go (hunk 3 above)
Comment (owais) [reply receiver/k8sclusterreceiver/internal/collection/clusteresourcequotas.go:29]: Happy to have an issue assigned to me and do it in a separate PR.

File: receiver/k8sclusterreceiver/README.md (hunk 2 above)
Comment (owais) [reply receiver/k8sclusterreceiver/README.md:283]: Sounds good.

File: receiver/k8sclusterreceiver/internal/collection/clusteresourcequotas.go (hunk 3 above)
Comment (jrcamp) [reply receiver/k8sclusterreceiver/internal/collection/clusteresourcequotas.go:29]: #4367

**Review by jrcamp — APPROVED** [review]

File: receiver/k8sclusterreceiver/factory.go [hunk 5]
```diff
@@ -56,7 +64,20 @@ func createMetricsReceiver(
...
-	return newReceiver(params.Logger, rCfg, consumer, k8sClient)
+
+	var osQuotaClient quotaclientset.Interface
+	switch rCfg.Distribution {
+	case distributionOpenShift:
+		osQuotaClient, err = rCfg.getOpenShiftQuotaClient()
+		if err != nil {
+			return nil, err
+		}
+	case distributionKubernetes:
```
Comment (anuraaga) [inline receiver/k8sclusterreceiver/factory.go:75]: I think a comment like `// Nothing to initialize` could help to make clear the empty block isn't a typo

**Review by anuraaga — APPROVED** [review]

**Pull request merged**